### PR TITLE
Fix deprecation for symfony/config 4.2+

### PR DESCRIPTION
**Pull request by bits4breakfast**
Heavily inspired by https://github.com/symfony/maker-bundle/pull/324

### CODE REVIEW

File: DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -41,8 +46,13 @@ public function getConfigTreeBuilder()
      */
     private function getVendorNode($name)
     {
-        $treeBuilder = new TreeBuilder();
-        $node = $treeBuilder->root($name);
+        $treeBuilder = new TreeBuilder($name);
+        if (\method_exists($treeBuilder, 'getRootNode')) {
+            $rootNode = $treeBuilder->getRootNode();
```
Comment (bits4breakfast) [inline DependencyInjection/Configuration.php]:
```suggestion
            $node = $treeBuilder->getRootNode();
```

File: DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -41,8 +46,13 @@ public function getConfigTreeBuilder()
...
+        } else {
+            // BC layer for symfony/config 4.1 and older
+            $rootNode = $treeBuilder->root($name);
```
Comment (bits4breakfast) [inline DependencyInjection/Configuration.php]:
```suggestion
            $node = $treeBuilder->root($name);
```

File: DependencyInjection/Configuration.php [hunk 3]
```diff
@@ -67,8 +77,13 @@ private function getVendorNode($name)
 
     private function getClassNode()
     {
-        $treeBuilder = new TreeBuilder();
-        $node = $treeBuilder->root('class');
+        $treeBuilder = new TreeBuilder('class');
+        if (\method_exists($treeBuilder, 'getRootNode')) {
+            $rootNode = $treeBuilder->getRootNode();
```
Comment (bits4breakfast) [inline DependencyInjection/Configuration.php]:
```suggestion
            $node = $treeBuilder->getRootNode();
```

File: DependencyInjection/Configuration.php [hunk 4]
```diff
@@ -67,8 +77,13 @@ private function getVendorNode($name)
...
+        } else {
+            // BC layer for symfony/config 4.1 and older
+            $rootNode = $treeBuilder->root('class');
```
Comment (bits4breakfast) [inline DependencyInjection/Configuration.php]:
```suggestion
            $node = $treeBuilder->root('class');
```

File: DependencyInjection/Configuration.php [hunk 5]
```diff
@@ -121,8 +136,13 @@ private function getClassNode()
 
     private function getUploadableNode()
     {
-        $treeBuilder = new TreeBuilder();
-        $node = $treeBuilder->root('uploadable');
+        $treeBuilder = new TreeBuilder('uploadable');
+        if (\method_exists($treeBuilder, 'getRootNode')) {
+            $rootNode = $treeBuilder->getRootNode();
```
Comment (bits4breakfast) [inline DependencyInjection/Configuration.php]:
```suggestion
            $node = $treeBuilder->getRootNode();
```

File: DependencyInjection/Configuration.php [hunk 6]
```diff
@@ -121,8 +136,13 @@ private function getClassNode()
...
+        } else {
+            // BC layer for symfony/config 4.1 and older
+            $rootNode = $treeBuilder->root('uploadable');
```
Comment (bits4breakfast) [inline DependencyInjection/Configuration.php]:
```suggestion
            $node = $treeBuilder->root('uploadable');
```

File: DependencyInjection/Configuration.php [hunk 7]
```diff
@@ -14,8 +14,13 @@ class Configuration implements ConfigurationInterface
      */
     public function getConfigTreeBuilder()
     {
-        $treeBuilder = new TreeBuilder();
-        $rootNode = $treeBuilder->root('stof_doctrine_extensions');
+        $treeBuilder = new TreeBuilder('stof_doctrine_extensions');
+        if (\method_exists($treeBuilder, 'getRootNode')) {
+            $rootNode = $treeBuilder->getRootNode();
```
Comment (bits4breakfast) [inline DependencyInjection/Configuration.php]:
```suggestion
            $node = $treeBuilder->getRootNode();
```

File: DependencyInjection/Configuration.php [hunk 8]
```diff
@@ -14,8 +14,13 @@ class Configuration implements ConfigurationInterface
...
+        } else {
+            // BC layer for symfony/config 4.1 and older
+            $rootNode = $treeBuilder->root('stof_doctrine_extensions');
```
Comment (bits4breakfast) [inline DependencyInjection/Configuration.php]:
```suggestion
            $node = $treeBuilder->root('stof_doctrine_extensions');
```

File: DependencyInjection/Configuration.php [hunk 9]
```diff
@@ -14,8 +14,13 @@ class Configuration implements ConfigurationInterface
      */
     public function getConfigTreeBuilder()
     {
-        $treeBuilder = new TreeBuilder();
-        $rootNode = $treeBuilder->root('stof_doctrine_extensions');
+        $treeBuilder = new TreeBuilder('stof_doctrine_extensions');
+        if (\method_exists($treeBuilder, 'getRootNode')) {
+            $node = $treeBuilder->getRootNode();
```
Comment (bits4breakfast) [inline DependencyInjection/Configuration.php]:
```suggestion
            $rootNode = $treeBuilder->getRootNode();
```

File: DependencyInjection/Configuration.php [hunk 10]
```diff
@@ -14,8 +14,13 @@ class Configuration implements ConfigurationInterface
...
+        } else {
+            // BC layer for symfony/config 4.1 and older
+            $node = $treeBuilder->root('stof_doctrine_extensions');
```
Comment (bits4breakfast) [inline DependencyInjection/Configuration.php]:
```suggestion
            $rootNode = $treeBuilder->root('stof_doctrine_extensions');
```

**Comment by bits4breakfast** [comment]
seems like Travis is failing while trying to process tests on PHP 5.3

**Comment by inverse** [comment]
Looks good :)

**Comment by bits4breakfast** [comment]
Hi @stof what is the process for submitting a PR? Thanks!

**Comment by stof** [comment]
thanks

**Comment by luispabon** [comment]
Thank you for this fix :+1: any word @stof on tagging a new release?

**Comment by spirit-q2** [comment]
@stof hi there! maybe you can release `1.3.1` just with this fix?

**Comment by cybernet** [comment]
it's been 10 months ... do we have to wait for how long ?

**Comment by tarlepp** [comment]
hmm, still no new release? is there some reason for that and if there is how could we help?

**Comment by inverse** [comment]
@stof ^^

**Comment by spirit-q2** [comment]
@tarlepp, @inverse take a look here: https://github.com/antishov/StofDoctrineExtensionsBundle/releases ;) it's a good fork.
thanks @antishov :+1:

**Comment by inverse** [comment]
@spirit-q2  while forking is a good thing I think it'll be better to ask @stof to be added as a contributor to this repo.

**Comment by spirit-q2** [comment]
@inverse sure thing. I agree with you 100%, but given that the question arose almost year ago (https://github.com/stof/StofDoctrineExtensionsBundle/pull/388#issue-242489474, on Jan 6 2019) I with you a good luck in waiting for response from @stof :trollface:

**Comment by maxhelias** [comment]
You need to read the issue about the project that is embed in this bundle : https://github.com/Atlantic18/DoctrineExtensions/issues/2033

**Comment by spirit-q2** [comment]
@stof if it happens that you will read this message :crossed_fingers: then I would like to be an contributor to this repo :)

**Pull request merged**